DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/mmhost/README.md

```diff
@@ -0,0 +1,335 @@
+> This practice ensures that your compiler, libraries, and tools are ready for development.
+>
+> Linux*:
+> - For system wide installations: `. /opt/intel/oneapi/setvars.sh`
+> - For private installations: ` . ~/intel/oneapi/setvars.sh`
+> - For non-POSIX shells, like csh, use the following command: `bash -c 'source <install-dir>/setvars.sh ; exec csh'`
+>
+> Windows*:
+> - `C:\Program Files(x86)\Intel\oneAPI\setvars.bat`
+> - Windows PowerShell*, use the following command: `cmd.exe "/K" '"C:\Program Files (x86)\Intel\oneAPI\setvars.bat" && powershell'`
+>
+> For more information on configuring environment variables, see [Use the setvars Script with Linux* or macOS*](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-linux-or-macos.html) or [Use the setvars Script with Windows*](https://www.intel.com/content/www/us/en/develop/documentation/oneapi-programming-guide/top/oneapi-development-environment-setup/use-the-setvars-script-with-windows.html).
+
+Use these commands to run the design, depending on your OS.
+
+### On a Linux* System
+This design uses CMake to generate a build script for GNU/make.
+
+1. Change to the sample directory.
+
+2. Configure the build system for the Agilex® 7 device family, which is the default.
+
+   ```
+   mkdir build
+   cd build
+   cmake ..
+   ```
+
+   > **Note**: You can change the default target by using the command:
+   >  ```
+   >  cmake .. -DFPGA_DEVICE=<FPGA device family or FPGA part number>
+   >  ```
+
+3. Compile the design using `make`.
+
+   | Compilation Type    | Command
+   |:---                 |:---
+   | FPGA Emulator       | `make fpga_emu`
+   | Optimization Report | `make report`
+   | FPGA Simulator      | `make fpga_sim`
+   | FPGA Hardware       | `make fpga`
+
+### On a Windows* System
+This design uses CMake to generate a build script for  `nmake`.
+
+1. Change to the sample directory.
+
+2. Configure the build system for the Agilex® 7 device family, which is the default.
+   ```
+   mkdir build
+   cd build
+   cmake -G "NMake Makefiles" ..
+   ```
+
+   > **Note**: You can change the default target by using the command:
+   >  ```
+   >  cmake -G "NMake Makefiles" .. -DFPGA_DEVICE=<FPGA device family or FPGA part number>
+   >  ```
+
+3. Compile the design using `nmake`.
+
+   | Compilation Type    | Command (Windows)
+   |:---                 |:---
+   | FPGA Emulator       | `nmake fpga_emu`
+   | Optimization Report | `nmake report`
+   | FPGA Simulator      | `nmake fpga_sim`
+   | FPGA Hardware       | `nmake fpga`
+
+   > **Note**: If you encounter any issues with long paths when compiling under Windows*, you may have to create your ‘build’ directory in a shorter path, for example c:\samples\build.  You can then run cmake from that directory, and provide cmake with the full path to your sample directory.
+
+## Examining the Reports
```

Comment:
You may also want to point to the example RTL block that we generate to showcase the differences at the RTL level of the iface.